### ISSUE TITLE
[Bug] anki freezes when switching to a different app during sync 

### ISSUE THREAD

**Issue by avshyz-gs** [ankidroid/Anki-Android#14197]
###### Reproduction Steps

1. Initiate a sync from the main screen (swiping down or clicking on the sync button) 
2. Switch to a different app
3. Switch back to anki

###### Expected Result
It might take several attempts, but after following up with the reproduction steps the app consistently freezes 
![Screenshot_20230806-125341.png](https://github.com/ankidroid/Anki-Android/assets/91832996/5e20163d-455b-4b5b-b015-3ae227b3b6de)

![Screenshot_20230806-125315.png](https://github.com/ankidroid/Anki-Android/assets/91832996/a07a9db5-8ea7-4b9d-b5db-3e52200c9c7c)

###### Actual Result
Anki freezes, and becomes unresponsive 

###### Debug info
AnkiDroid Version = 2.16.2

Android Version = 13

ProductFlavor = play

Manufacturer = Google

Model = Pixel 6a

Hardware = bluejay

Webview User Agent = Mozilla/5.0 (Linux; Android 13; Pixel 6a Build/TQ3A.230705.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/115.0.5790.138 Mobile Safari/537.36

ACRA UUID = 0a6738e2-baa5-4e84-8f59-ddd1b32ff4dd

New schema = false

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by avshyz-gs** [comment]
It's worth noting I've never had this issue before the last update

**Comment by user1823** [comment]
Similar to https://github.com/ankidroid/Anki-Android/issues/13764

**Comment by BrayanDSO** [comment]
Had this as well ocasionally

**Comment by tbm** [comment]
I see the same after the recent upgrade. This never happened before.

**Comment by Seagist** [comment]
Same. Just minimizing the app is enough for this to happen.

**Comment by BrayanDSO** [comment]
My guess it that this is related to Android's current way of dealing with background tasks, since it works considerably better in an Android 10 device I have than in my Android 13 phone.

In 2.17, it should be fixed when #12468 is done, but I fear that this may take some time.

**Comment by dEhiN** [comment]
I came here to report the same issue! In my case, I don't switch to another app but either swipe up fully to go to the home (using phone's gestures feature) or click the power button to turn off and lock the screen.

When I unlock the screen or click on Anki from the home screen, it is frozen on the sync process. I haven't received any errors but I can't do anything, the app is unresponsive. I have to force close it and reopen.

Here's a screenshot of what I see:
![Screenshot_20230808-152007](https://github.com/ankidroid/Anki-Android/assets/26848686/32e15365-aa03-49d6-ae48-06b46a3250e0)

**Comment by pratt360** [comment]
Just wanted to pop in and say I've been having this same issue.

**Comment by jgscherber** [comment]
Also experiencing this issue. Repro step is to initialize sync and then immediately hit home button. I tried disabling battery optimization for AnkiDroid and issue remained

---

AnkiDroid Version = 2.16.5 (953b9bd879269910ee962b520da9705336d8bc2d)

Android Version = 13

ProductFlavor = play

Manufacturer = samsung

Model = SM-G981U1

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 13; SM-G981U1 Build/TP1A.220624.014; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/116.0.0.0 Mobile Safari/537.36

ACRA UUID = 03434583-cd6c-4869-8ed2-d8e7712382df

New schema = false

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true

**Comment by azoo95** [comment]
I am also hitting this issue for months since the big update. 

https://github.com/ankidroid/Anki-Android/assets/5181100/0354185b-2f6e-4cf9-9258-e6fea1406404

**Comment by david-allison** [comment]
@azoo95 still reproducible in the 2.17 beta?

**Comment by azoo95** [comment]
@david-allison Sorry not to provide the version. It was 2.16.5 official. Let me try again when I get 2.17 soon.
